src/Runtime/Distributions/Automata/Automaton.EpsilonClosure.cs

```diff
@@ -32,7 +32,7 @@ public class EpsilonClosure
             /// <summary>
             /// The list of the states in the closure.
             /// </summary>
-            private readonly List<Pair<State, Weight>> weightedStates = new List<Pair<State, Weight>>(DefaultStateListCapacity);
+            private readonly List<ValueTuple<State, Weight>> weightedStates = new List<ValueTuple<State, Weight>>(DefaultStateListCapacity);
```

Comment:
This code should be using a custom struct WeightedState with fields "State" and "Weight"